### PR TITLE
WCOR-1257 Responsive image styles for embedded images with demonstration WIP

### PR DESCRIPTION
**Pull request by joshuami**
### ODE XX

**Use/Fix Case** 

Responsive images will allow for a wide range of embedded image formats that deliver an appropriately sized image for many display sizes. These images should also play nicely whether they are embedded with alignment (float) or via a grid (Bootstrap columns).

**Implementation Details**

- Configure the basic image styles and responsive image styles needed for media library.
- Add Imagemagick configuration for better image optimization.
- Adds Focal Point cropping for follow up story using featured images.
- Update style sheets to clean up embedded media styles and better document what the styles accomplish.
- Create a demo page on the CD for testing

**Dependencies**

- focal_point, image_effects, imagemagick

**Test Plan**

- [ ] First step of the test plan is for @joshuami to create a demo page for testing. It will be important not to edit any existing images on the site as that would update the saved image styles for that image in the S3 bucket.

**Post Deployment**
- [ ] When we merge this PR, we may need to flush the image styles so that new images will be generated for the responsive image styles configured. (Or we may not as I reused some image styles that were already in place.)
- [ ] The demonstration will need to be recreated on dev so that editors have examples to review.

### CODE REVIEW

**Comment by sharstine** [comment]
@joshuami - Some adjustments may be needed before QA passes.

### Feedback on image examples

The Elephant Image Examples page is very helpful. The writeup is clear to me, but most editors won't be familiar with terms like Bootstrap, floats, pixel density, empty paragraphs, and CKEditor. It's a great reference page and we can create training materials that use simpler terms, so I think it's OK to leave the reading level as is. However, there are some things I'd suggest changing:

- Remove the word "Elephant" from the page title if editors will be referring to this page.

- "... can use three differen[ce] display modes ..." -- Typo

- "Not all combinations of embed options are supported." -- This statement was unclear to me until I started testing and realized that narrow/center isn't a valid combination, for example. I think editors may be confused by this statement as well, because we're offering layout examples and then saying that not all options (presumably the options we're offering as examples) are supported. Could we reword it to something like this: "Other variations of image layouts are not supported."

- "... high density displays such using ..." -- Is there a word missing here?

- "On small displays, such as mobile devices. Images aligned left or right will appear before the text as a full-width image." -- I think this is meant to be one sentence with a comma after "devices."

- If the audience of this page is editors, I don't think the technical language about captions will make sense. Perhaps we add a note saying that captions can be used, but the DigEx team must help you add them for now.

### Feedback on image display modes

- **Narrow display / left align / with or without caption** - The image is _right-aligned_ in large viewports, despite choosing left alignment. See the very last image and third-to-last image on /housing/scott-qa-test-page on ODE281.

- **Narrow display / center align / with caption** - It works fine in small viewports, but in larger viewports the image is actually _left-aligned_, not centered. However, I can't think of a valid use case for this display/align combination. See the second-to-last image on /housing/scott-qa-test-page on ODE281. I think it's OK if we do nothing to address this invalid edge case.

**Comment by joshuami** [comment]
@sharstine , thank you for the thorough review. I sheepishly must admit that I missed a couple of copy and paste errors on my narrow display mode CSS. I can get that fixed right up. It should even account for the version that we don't expect: narrow + centered.

All the content-related feedback was great as well. I definitely wrote it as a primer for the DigX team rather than for end users. (I was also trying to fill enough space to avoid some of the weird things that happen with floated images. 😄 )

The title elephant was just to make it easy to find for QA. When we create this on Dev, I think we should spend a little time turning that page into a style guide under a new group for "Website Style Guide and Components". It would be nice to the the elephant test page from I.F. into that format as well. I'll write up a story to start pulling that together.

**Comment by joshuami** [comment]
@sharstine , this is ready for another round of QA when the build completes. I think you'll lose your test content, but you can recreate it by pasting this into a new page:

```
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque pharetra sed mi et accumsan. Suspendisse potenti. Cras ut lorem pulvinar, euismod lacus vel, tempor eros. Fusce porttitor tortor in venenatis malesuada. Pellentesque egestas iaculis magna, a aliquam mauris luctus nec. Etiam sed aliquam est. Vivamus et rhoncus tellus.</p>

<drupal-media data-align="center" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="a523d183-ff9c-4de1-968d-896f9406df76"></drupal-media>

<p>Sed sit amet dignissim neque. Aliquam placerat tempus urna, at laoreet mi facilisis quis. Fusce imperdiet eget nibh vel iaculis. Sed dolor mauris, consequat ac lectus quis, volutpat convallis lorem. Praesent pulvinar, neque ut porta ultricies, tellus nibh congue dui, sit amet maximus quam mi vitae ante.</p>

<drupal-media data-align="right" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="a523d183-ff9c-4de1-968d-896f9406df76" data-view-mode="half_width"></drupal-media>

<p>Nullam gravida convallis tellus et ullamcorper.</p>

<p>Donec accumsan lorem sit amet commodo cursus. Suspendisse porta pretium est, at gravida sapien fermentum ac. Suspendisse venenatis lobortis pellentesque. Vivamus sit amet velit odio. Morbi sed arcu nunc.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>In purus elit, finibus id nisi quis, dignissim bibendum enim. Phasellus non lorem interdum, aliquet nisi eget, egestas ipsum.</p>

<p>Fusce lorem neque, sodales eu libero eget, viverra interdum ligula. Integer porttitor gravida ex, vel euismod nisi fringilla sed. Donec quis elementum dolor, nec iaculis lorem. Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<p>Fusce lorem neque, sodales eu libero eget, viverra interdum ligula. Integer porttitor gravida ex, vel euismod nisi fringilla sed. Donec quis elementum dolor, nec iaculis lorem. Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<p>Fusce lorem neque, sodales eu libero eget, viverra interdum ligula. Integer porttitor gravida ex, vel euismod nisi fringilla sed. Donec quis elementum dolor, nec iaculis lorem. Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<p>Fusce lorem neque, sodales eu libero eget, viverra interdum ligula. Integer porttitor gravida ex, vel euismod nisi fringilla sed. Donec quis elementum dolor, nec iaculis lorem. Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<div class="row two-col">
<div class="col-12 col-sm-6 col-md-6 col-first">
<p>Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<p>Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>

<p>Vestibulum quis arcu ligula. Suspendisse ac justo consequat, eleifend ligula quis, efficitur diam.</p>
</div>

<div class="col-12 col-sm-6 col-md-6 col-second">
<drupal-media data-align="center" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="a523d183-ff9c-4de1-968d-896f9406df76"></drupal-media>
</div>
</div>

<p>Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.&nbsp;Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.&nbsp;Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.&nbsp;Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.</p>

<div class="row three-col">
<div class="col-12 col-sm-4 col-md-4 col-first">
<p>Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.</p>

<p>Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.</p>
</div>

<div class="col-12 col-sm-4 col-md-4 col-second">
<drupal-media data-align="center" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="a523d183-ff9c-4de1-968d-896f9406df76"></drupal-media>
</div>

<div class="col-12 col-sm-4 col-md-4 col-third">
<p>Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.</p>

<p>Praesent in maximus neque. Maecenas elit sapien, mattis sed ex sed, elementum posuere eros. Aenean ferment.</p>
</div>
</div>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<drupal-media data-align="right" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="5e7a0b82-5ce1-4a50-8197-9de517696f59" data-view-mode="narrow"></drupal-media>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<drupal-media data-align="left" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="5e7a0b82-5ce1-4a50-8197-9de517696f59" data-view-mode="narrow"></drupal-media>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<drupal-media data-align="center" data-caption="Donec interdum arcu nunc, in hendrerit mi sodales sit amet. Etiam in blandit risus. Vestibulum iaculis felis eu nibh porttitor dapibus. Vestibulum tincidunt eget ante vitae ornare. Donec libero elit, facilisis id accumsan in, elementum ac nunc." data-entity-type="media" data-entity-uuid="5e7a0b82-5ce1-4a50-8197-9de517696f59" data-view-mode="narrow"></drupal-media>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<drupal-media data-align="left" data-entity-type="media" data-entity-uuid="5e7a0b82-5ce1-4a50-8197-9de517696f59" data-view-mode="narrow"></drupal-media>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>Suspendisse faucibus efficitur lorem quis ullamcorper. Suspendisse est velit, malesuada nec mollis eu, mattis ut enim. Nunc condimentum efficitur ipsum sit amet tincidunt. Curabitur in libero urna. Curabitur eleifend augue turpis, vitae vulputate enim consequat sit amet.</p>

<p>&nbsp;</p>
```
